src/sessions.ts

```diff
@@ -729,25 +733,25 @@ function endTransaction(session: ClientSession, commandName: string, callback: C
     }
 
     session.transaction.transition(TxnState.TRANSACTION_COMMITTED);
-    if (e) {
+    if (error instanceof MongoError) {
       if (
-        e instanceof MongoNetworkError ||
-        e instanceof MongoWriteConcernError ||
-        isRetryableError(e) ||
-        isMaxTimeMSExpiredError(e)
+        error instanceof MongoNetworkError ||
+        error instanceof MongoWriteConcernError ||
+        isRetryableReadError(error) || // TODO is commit a write? probably, double check its retry rules
```

Comment:
any context on this TODO? do we need to figure out if this check also needs to check for retryable write now that they're split up?

Reply:
Removed the TODO and simplified the logic here